test/numa/run.sh

```diff
@@ -0,0 +1,482 @@
+                echo "    verify command output: $COMMAND_OUTPUT"
+                exit 1
+        }
+    done
+}
+
+delete() { # script API
+    # Usage: delete PARAMETERS
+    #
+    # Run "kubectl delete PARAMETERS".
+    vm-command "kubectl delete $*" || {
+        command-error "kubectl delete failed"
+    }
+}
+
+create() { # script API
+    # Usage: [VAR=VALUE] create TEMPLATE_NAME
+    #
+    # Create n instances from TEMPLATE_NAME.yaml.in, copy each of them
+    # from host to vm, kubectl create -f them, and wait for them
+    # becoming Ready.
+    #
+    # Parameters:
+    #   TEMPLATE_NAME: the name of the template without extension (.yaml.in)
+    #
+    # Optional parameters (VAR=VALUE):
+    #   wait: condition to be waited for (see kubectl wait --for=condition=).
+    #         If empty (""), skip waiting. The default is wait="Ready".
+    #   wait_t: wait timeout. The default is wait_t=60s.
+    template_name=$1.yaml.in
+    local template_kind
+    template_kind=$(awk '/kind/{print tolower($2)}' < "$template_name")
+    local wait=${wait-Ready}
+    local wait_t=${wait_t-60s}
+    if [ -z "$n" ]; then
+        local n=1
+    fi
+    if [ ! -f "$template_name" ]; then
+        error "error creating \"$1\": missing template ${template_name}"
+    fi
+    for _ in $(seq 1 $n); do
+        kind_count[$template_kind]=$(( ${kind_count[$template_kind]} + 1 ))
+        local NAME="${template_kind}$(( ${kind_count[$template_kind]} - 1 ))" # the first pod is pod0
+        eval "echo -e \"$(<"${template_name}")\"" > "$NAME.yaml"
+        host-command "scp $NAME.yaml $VM_SSH_USER@$VM_IP:" || {
+            command-error "copying $NAME.yaml to VM failed"
+        }
+        vm-command "cat $NAME.yaml"
+        vm-command "kubectl create -f $NAME.yaml" || {
+            command-error "kubectl create error"
+        }
+        if [ "x$wait" != "x" ]; then
+            speed=1000 vm-command "kubectl wait --timeout=${wait_t} --for=condition=${wait} ${template_kind}/$NAME" >/dev/null 2>&1 || {
+                command-error "waiting for ${template_kind} \"$NAME\" to become ready timed out"
+            }
+        fi
+    done
+}
+
+help() { # script API
+    # Usage: help [FUNCTION|all]
+    #
+    # Print help on all functions or on the FUNCTION available in script.
+    awk -v f="$1" \
+        '/^[a-z].*script API/{split($1,a,"(");if(f==""||f==a[1]||f=="all"){print "";print a[1]":";l=2}}
+         !/^    #/{l=l-1}
+         /^    #/{if(l>=1){split($0,a,"#"); print "   "a[2]; if (f=="") l=0}}' <<<"$script_source"
+}
+
+### End of user code helpers
+
+test-user-code() {
+    vm-command-q "kubectl get pods 2>&1 | grep -q NAME" && vm-command "kubectl delete pods --all --now"
+    ( eval "$code" ) || {
+        TEST_FAILURES="${TEST_FAILURES} test script failed"
+    }
+}
+
+# Validate parameters
+mode=$1
+user_script_file=$2
+vm=${vm-"crirm-test-numa"}
+cri_resmgr_cfg=${cri_resmgr_cfg-"cri-resmgr-memtier.cfg"}
```

Comment:
Defaulting to ${SCRIPT_DIR}/cri-resmgr-memtier.cfg would be more convenient. Then one neither would need to `cd test/numa` nor specify `cri_resmgr_cfg=test/numa/cri-resmgr-memtier.cfg` when running `./test/numa/run.sh interactive` for interactive testing.

Reply:
Thanks, fixed!